basic/plot_functions.h

```diff
@@ -0,0 +1,43 @@
+
```

Comment:
Few things about this file:

1) Make sure to add `#pragma once` at the top
2) All of the functions in this header will be publicly visible to the linker and cause linking errors if included in multiple files. It would be nice to have them templated too - so see 3).
3) Make all of the functions here templated, just so they technically work with `float` too (and can also get around the multiple definitions errors from being in a header)